src/libraries/System.Private.CoreLib/src/System/IO/BinaryWriter.cs

```diff
@@ -15,32 +15,30 @@ namespace System.IO
     //
     public class BinaryWriter : IDisposable, IAsyncDisposable
     {
+        private const int MaxArrayPoolRentalSize = 1024 * 1024; // ArrayPool<T>.Shared allocates beyond this point
```

Comment:
This should not really depend on ArrayPool implementation details. It would be better to set this to size where we start to see diminishing results. I would expect it to be like 64kB.

Reply:
Also, very large buffers tend to not work that well since they do not fit into processor cache.

Reply:
Would it make sense to refactor the Stream copy buffer size const out into an internal field, then reference it from here? That would provide a single place to look across our code when it needs to figure out a good default buffer size.

https://github.com/dotnet/runtime/blob/8a52f1e948b6f22f418817ec1068f07b8dae2aa5/src/libraries/System.Private.CoreLib/src/System/IO/Stream.cs#L122

Reply:
I am not sure. It is not clear whether the `DefaultCopyBufferSize` is actually a good default buffer size.

Reply:
The original justification for 81920 was that it is right under default LOH threshold and good for GC. This argument does not hold with ArrayPool that was not used originally. ArrayPool will round it up to the next power of 2, so 81920 will turn into 128k that is right above LOH threshold... .

Reply:
FWIW, some time last year I tried decreasing the size so it would be back under the LOH threshold even after the pool rounded up,  but there were quite measurable regressions for certain operations on microbenchmarks due to the much smaller buffer size,  so I left it as is until we had a pressing scenario highlighting it was worth a change.